src/consul.rs

```diff
@@ -0,0 +1,452 @@
+use anyhow::{bail, Context, Result};
+use base64::prelude::{Engine as _, BASE64_STANDARD};
+use reqwest::{StatusCode, Url};
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::HashMap,
+    future::Future,
+    time::{Duration, SystemTime},
+};
+use tokio::{
+    task::JoinHandle,
+    time::{interval, MissedTickBehavior},
+};
+use tokio_util::sync::CancellationToken;
+use uuid::Uuid;
+
+use crate::dns_trait::DnsType;
+
+const CONSUL_STORE_KEY: &str = "dns_records_lock";
```

Comment:
We probably want this key to be the same as the name of the project to make it obvious to anyone looking at the keys where they're from.

Reply:
Used `consul_external_dns` as store key